### PR TITLE
Refine login, signup and password reset page

### PR DESCRIPTION
**Pull request by megascrapper**
The login, signup and password reset pages has been rewritten using MUI grids and React state controls, as per suggestions by @VenuraWe and @Brittank88.

### CODE REVIEW

File: src/content/login/index.tsx [hunk 1]
```diff
@@ -1,70 +1,242 @@
...
+            container
+            spacing={1}
+            alignItems="center"
+            justifyContent="center"
+            textAlign="center"
+            component="form"
+            onSubmit={handleSubmit}
+        >
+            <Grid item xs={12}>
+                <TextField
```
Comment (JMSS-Unknown) [inline src/content/login/index.tsx]: Pass value and onChange handler here

File: src/content/login/index.tsx [hunk 2]
```diff
@@ -1,70 +1,242 @@
...
+ */
+const LoginForm = () => {
+    const [passwordVisible, setPasswordVisible] = useState(false);
+
+    const [rememberMe, setRememberMe] = useState(false);
+
+    const [usernameIR, passwordIR] = [
+        useRef<HTMLInputElement>(null),
+        useRef<HTMLInputElement>(null),
+    ];
```
Comment (JMSS-Unknown) [inline src/content/login/index.tsx]: It is not recommended to use refs, instead make the components controlled. See https://reactjs.org/docs/forms.html#controlled-components (Note the docs are outdated and use class components, but the same logic still applies)

File: src/content/login/index.tsx [hunk 3]
```diff
@@ -1,70 +1,242 @@
...
+    const [username, setUsername] = useState("");
+    const [password, setPassword] = useState("");
+    const [usernameError, setUsernameError] = useState(false);
+    const [passwordError, setPasswordError] = useState(false);
+
+    const handleSubmit = (e: FormEvent<HTMLFormElement>) => {
+        e.preventDefault();
+        setUsername(usernameIR.current?.value || "");
+        setPassword(passwordIR.current?.value || "");
```
Comment (JMSS-Unknown) [inline src/content/login/index.tsx]: This should be done on input change

**Review by JMSS-Unknown — CHANGES_REQUESTED** [review]
I left some comments about making the inputs controlled as uncontrolled inputs should be avoided. Also if we are going to have lots of forms it may be worth using `react-hook-form`.

File: src/content/login/index.tsx (hunk 1 above)
Comment (ghost) [reply src/content/login/index.tsx]: I love the idea of using `react-hook-form`, seems like it could be a good way to simplify the form creation process.

**Comment by ghost** [comment]
PR unnecessary due to #13.

**Pull request closed without merging**